### PR TITLE
feat(module:calendar): sync some changes from ant-design (#2769)

### PR DESCRIPTION
**Pull request by zhytang**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #2769 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/calendar/demo/select.ts [hunk 1]
```diff
@@ -4,9 +4,20 @@ import { Component } from '@angular/core';
   selector: 'nz-demo-calendar-select',
   template: `
     <nz-alert nzMessage="Your selected date: {{ selectedValue | date:'yyyy-MM-dd' }}"></nz-alert>
-    <nz-calendar [(ngModel)]="selectedValue"></nz-calendar>
+    <nz-calendar [(ngModel)]="selectedValue"
+      (nzSelecgChange)="selectChange($event)"
```
Comment (vthinkxie) [inline components/calendar/demo/select.ts]: nzSelecgChange -> nzSelectChange

File: components/calendar/nz-calendar.component.ts [hunk 2]
```diff
@@ -108,22 +113,26 @@ export class NzCalendarComponent implements ControlValueAccessor, OnInit {
...
 
   onDateSelect(date: Date): void {
     this.updateDate(date);
+    this.nzSelectChange.emit(date);
   }
 
   onYearSelect(year: number): void {
     const date = setYear(this.activeDate, year);
     this.updateDate(date);
+    this.nzPanelChange.emit({'date': date, 'mode': this.nzMode});
```
Comment (vthinkxie) [inline components/calendar/nz-calendar.component.ts]: should be `this.nzSelectChange.emit(date);`

File: components/calendar/nz-calendar.component.ts [hunk 3]
```diff
@@ -108,22 +113,26 @@ export class NzCalendarComponent implements ControlValueAccessor, OnInit {
...
   }
 
   onMonthSelect(month: number): void {
     const date = setMonth(this.activeDate, month);
     this.updateDate(date);
+    this.nzPanelChange.emit({'date': date, 'mode': this.nzMode});
```
Comment (vthinkxie) [inline components/calendar/nz-calendar.component.ts]: should be this.nzSelectChange.emit(date);

Comment (zhytang) [reply components/calendar/nz-calendar.component.ts]: I have fixed it & modified nz-calendar.spec.ts

File: components/calendar/nz-calendar.component.ts [hunk 4]
```diff
@@ -108,22 +113,26 @@ export class NzCalendarComponent implements ControlValueAccessor, OnInit {
     this.calculateActiveMonth();
   }
 
-  onModeChange(mode: 'month'|'year'): void {
+  onModeChange(mode: ModeType): void {
     this.nzModeChange.emit(mode);
+    this.nzPanelChange.emit({'date': this.activeDate, 'mode': mode});
```
Comment (vthinkxie) [inline components/calendar/nz-calendar.component.ts]: 组合 nzModeChange 和 nzValueChange 就可以得到 nzPanelChange 的内容，为什么还要再增加 nzPanelChange 这个 output

File: components/calendar/nz-calendar.component.ts [hunk 5]
```diff
@@ -108,22 +113,26 @@ export class NzCalendarComponent implements ControlValueAccessor, OnInit {
...
   onYearSelect(year: number): void {
     const date = setYear(this.activeDate, year);
     this.updateDate(date);
+    this.nzSelectChange.emit(date);
   }
 
   onMonthSelect(month: number): void {
     const date = setMonth(this.activeDate, month);
     this.updateDate(date);
+    this.nzSelectChange.emit(date);
```
Comment (vthinkxie) [inline components/calendar/nz-calendar.component.ts]: nzSelectChange 和 nzValueChange 的区别是什么

File: components/calendar/nz-calendar.component.ts (hunk 4 above)
Comment (zhytang) [reply components/calendar/nz-calendar.component.ts]: 为了和ant-design同步，这个issue不是这个意思？

**Pull request merged**